internal/exec/exec.go

```diff
@@ -85,7 +89,17 @@ func (r *Request) execSelections(ctx context.Context, sels []selected.Selection,
 				execFieldSelection(ctx, r, s, f, &pathSegment{path, f.field.Alias}, true)
 			}(f)
 		}
-		wg.Wait()
+		// close a signal channel once the wait group is complete
+		c := make(chan struct{})
```

Comment:
nit: can we name this something like `done`? No hard and fast rules but it's slightly more canonical and readable.